component/src/main/java/org/wso2/extension/siddhi/io/websocket/sink/WebSocketSink.java

```diff
@@ -0,0 +1,204 @@
+/*
+ *  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ *
+ */
+
+package org.wso2.extension.siddhi.io.websocket.sink;
+
+import org.wso2.extension.siddhi.io.websocket.util.WebSocketClientConnectorListener;
+import org.wso2.extension.siddhi.io.websocket.util.WebSocketProperties;
+import org.wso2.extension.siddhi.io.websocket.util.WebSocketUtil;
+import org.wso2.siddhi.annotation.Example;
+import org.wso2.siddhi.annotation.Extension;
+import org.wso2.siddhi.annotation.Parameter;
+import org.wso2.siddhi.annotation.util.DataType;
+import org.wso2.siddhi.core.config.SiddhiAppContext;
+import org.wso2.siddhi.core.exception.ConnectionUnavailableException;
+import org.wso2.siddhi.core.exception.SiddhiAppCreationException;
+import org.wso2.siddhi.core.stream.output.sink.Sink;
+import org.wso2.siddhi.core.util.config.ConfigReader;
+import org.wso2.siddhi.core.util.transport.DynamicOptions;
+import org.wso2.siddhi.core.util.transport.OptionHolder;
+import org.wso2.siddhi.query.api.definition.StreamDefinition;
+import org.wso2.transport.http.netty.contract.websocket.HandshakeFuture;
+import org.wso2.transport.http.netty.contract.websocket.WebSocketClientConnector;
+import org.wso2.transport.http.netty.contract.websocket.WsClientConnectorConfig;
+import org.wso2.transport.http.netty.contractimpl.HttpWsConnectorFactoryImpl;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.ByteBuffer;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * {@code WebsocketSink } Publishing the siddhi events to the WebSocket server.
+ */
+
+@Extension(
+        name = "websocket",
+        namespace = "sink",
+        description = "A Siddhi application can be configured to publish events via the Websocket transport by " +
+                "adding the @Sink(type = ‘Websocket’) annotation at the top of an event stream definition.",
```

Comment:
@Sink(type = ‘**Websocket**’) or @Sink(type = ‘**websocket**’) ?

Reply:
it should be websocket.

Reply:
fixed in a389891